pkg/certrotation/target.go

```diff
@@ -0,0 +1,141 @@
+package certrotation
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/certs"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/cert"
+)
+
+// TargetRotation rotates a key and cert signed by a CA. It creates a new one when 80%
+// of the lifetime of the old cert has passed, or the CA used to signed the old cert is
+// gone from the CA bundle.
+type TargetRotation struct {
+	Namespace     string
+	Name          string
+	Validity      time.Duration
+	HostNames     []string
+	Lister        corev1listers.SecretLister
+	Client        corev1client.SecretsGetter
+	EventRecorder events.Recorder
+}
+
+func (c TargetRotation) EnsureTargetCertKeyPair(signingCertKeyPair *crypto.CA, caBundleCerts []*x509.Certificate) error {
```

Comment:
where is this method referenced? i can clearly see the purpose and the implementation of this method but didnt find anywhere using it.